django/conf/__init__.py

```diff
@@ -280,5 +310,20 @@ def __repr__(self):
             'cls': self.__class__.__name__,
         }
 
+    @property
+    def SECRET_KEY(self):
+        try:
+            return self.SECRET_KEYS[0]
+        except TypeError:
+            return ''
+
+    @SECRET_KEY.setter
+    def SECRET_KEY(self, value):
+        self.SECRET_KEYS = [value]
+
+    @SECRET_KEY.deleter
+    def SECRET_KEY(self):
+        del self.SECRET_KEYS
+
```

Comment:
Do we need setters/getters and extra code in `is_overridden()`? Maybe it enough to set `SECRET_KEYS` in `Settings.__init__()`, e.g.

```python
class Settings:
    def __init__(self, settings_module):
        ...
        if self.is_overridden('SECRET_KEY'):
            if self.is_overridden('SECRET_KEYS'):
                raise ImproperlyConfigured(
                    'SECRET_KEY and SECRET_KEYS are mutually exclusive.'
                )
            setattr(self, 'SECRET_KEYS', [self.SECRET_KEY])
```
I would add `SECRET_KEYS` to the `tuple_settings`.

Reply:
> Do we need setters/getters ... ?

So this is the question in https://github.com/django/django/pull/13850/commits/2f1d3a397e621a1a0f82a91bbd00e7c950599dc6 — does it look better (is it more robust) having one source of truth for the two settings, vs trying to keep them in sync? (Could go either way here, but pushed this to get opinions.) 

The property removes the need to do the same (flaky?) dance in `override_settings` — and note `UserSettingsHolder` wasn't covered at all before. 

> ... and extra code in is_overridden()

That was just the minimum I needed to get the tests to pass. Maybe/likely there's a better way of handling that.

Reply:
I'm leaning towards @felixxm idea of having a single point of truth and not needing to keep them in sync. Keeping something in sync is always hard and error prone. But if there's only one way to get to `settings.SECRET_KEY` we can encapsulate all logic in there.

Reply:
Hmmm. This was (≈) the original idea, but it was difficult to handle all the cases: providing SECRET_KEY, providing SECRET_KEYS, making sure the values were sane when mapping between the options (in `Settings`, in `UserSettingsHolder`, in `override_settings()`) — it was precisely the issue of keeping two sources of truth in sync. 

That's why I went for the `@property` approach. `SECRET_KEY` defers then (always) to `SECRET_KEYS`. `override_settings` is then not altered at all (which is a win.) I thought the diff in 2f1d3a3 made it much cleaner, as well as having it work for `UserSettingsHolder` along with `Settings`. 

There **is** a bit of duplication currently between `Settings` and `UserSettingsHolder` that we **could** move into a mixin. (As ever, we could go either way on what's cleaner there. 🤷‍♂️)

This is the main bit to pin down with @felixxm next week.